src/frontend/src/handler/util.rs

```diff
@@ -140,16 +141,29 @@ fn timestamptz_to_string_with_session_data(
 fn to_pg_rows(
     column_types: &[DataType],
     chunk: DataChunk,
-    format: bool,
+    formats: &[Format],
     session_data: &StaticSessionData,
 ) -> RwResult<Vec<Row>> {
+    // Invariant check
+    if !formats.is_empty() && formats.len() != 1 && formats.len() != column_types.len() {
+        return Err(ErrorCode::InternalError(format!(
+            "format codes length {} is not 0, 1 or equal to column length {}",
+            formats.len(),
+            column_types.len()
+        ))
+        .into());
+    }
+
```

Comment:
Superseded by `FormatIterator::new` below? Or is this just `column_types.len() == chunk.dimension()`

Reply:
I think here can be remove.